File: languages/go/01-foundations/scope-and-lifetime-basics/example/session-summary.go
// This helper example focuses on isolating one scope-sensitive helper so lifetime differences stay visible.

// This extra example extends scope and lifetime basics with a session summary.

package main

import "fmt"

// Keep this helper separate so the main example can focus on the larger idea without extra noise.
const bonusPoints = 5
const passingScore = 70

func summarizeAttempt(name string, score int) (string, bool) {
	effectiveScore := score + bonusPoints
	passed := effectiveScore >= passingScore
	status := "needs review"
	if passed {
		status = "passed"
	}

	return fmt.Sprintf("%s: raw=%d, effective=%d, status=%s", name, score, effectiveScore, status), passed
}

func main() {
	attempts := []struct {
		name  string
		score int
	}{
		{"Ana", 64},
		{"Bruno", 71},
		{"Carla", 58},
	}

	passedCount := 0
	for _, attempt := range attempts {
		line, passed := summarizeAttempt(attempt.name, attempt.score)
		fmt.Println(line)
		if passed {
			passedCount++
		}
	}

	fmt.Printf("Students above the target: %d\n", passedCount)
}

File: languages/python/01-foundations/scope-and-lifetime-basics/example/session-summary.py
# This helper example focuses on isolating one scope-sensitive helper so lifetime differences stay
# visible.

# This extra example extends scope and lifetime basics with a session summary.

BONUS_POINTS = 5
PASSING_SCORE = 70


# Keep this helper separate so the main example can focus on the larger idea without extra noise.
def summarize_attempt(name, score):
    effective_score = score + BONUS_POINTS
    passed = effective_score >= PASSING_SCORE
    status = "passed" if passed else "needs review"
    return f"{name}: raw={score}, effective={effective_score}, status={status}", passed


attempts = [("Ana", 64), ("Bruno", 71), ("Carla", 58)]
passed_count = 0

for student_name, raw_score in attempts:
    line, passed = summarize_attempt(student_name, raw_score)
    print(line)
    if passed:
        passed_count += 1

print(f"Students above the target: {passed_count}")
